lib/models.js

```diff
@@ -1224,18 +1453,21 @@ class Value {
       clone._constraints.push(constraint.clone());
     }
     if (this._inheritance) {
-      clone._inheritance = this.inheritance;
+      clone._inheritance = this._inheritance;
     }
     if (this._inheritedFrom) {
-      clone._inheritedFrom = this.inheritance;
+      clone._inheritedFrom = this._inheritedFrom;
+    }
+    if (this._constraintHistory) {
+      clone._constraintHistory = this._constraintHistory;
```

Comment:
this._constraintHistory is an object, so we should mostly likely be cloning it instead of a direct reference, e.g. `this._constraintHistory.clone()'